### DISCUSSION TITLE
Why not calculate more metrics during training loop?

### DISCUSSION THREAD

**Discussion by imteekay** [deep-learning-for-biology/notebooks#8] [Q&A] [answered]
Is there any reason why not calculate the metrics (accuracy, recall, precision, auprc, auroc) for the training step, only for the validation set?

**Comment by cnjr2** [comment, marked as answer]
Hi @imteekay - You can certainly calculate these metrics for the training set and get an impression of how they vary during model training. However, the [`calculate_metrics`](https://github.com/deep-learning-for-biology/dlfb/blob/d9b230aed966eb9be5fd394079e2ed78ecab0e22/src/dlfb/proteins/train.py#L88) in this chapter relies on metrics implemented in `sklearn` which are not `jit`table. So in order to run the training loop with `jit`, we skipped the metrics beyond the required loss. In some chapters we have implemented additional metrics to be compatible with `jit` (more difficult for some), so you could use them here too. Alternatively, as it is a small model, you can simply disable `jit` here and then use the calculate metrics function with all the metrics.

**Reply by imteekay** [reply]
Thanks, that's helpful @cnjr2
